libraries/fc/include/fc/static_variant.hpp

```diff
@@ -14,587 +14,86 @@
-        return impl::storage_ops<R, 0, Types...>::apply(_tag, storage, std::forward<visitor>(v));
-    }
-
-
-    template<typename R, typename visitor>
-    auto visit(visitor&& v) const & {
-        return impl::storage_ops<R, 0, Types...>::apply(_tag, storage, std::forward<visitor>(v));
-    }
-
-    template<typename R, typename visitor>
-    auto visit(visitor&& v) && {
-        return impl::storage_ops<R, 0, Types...>::apply_rvalue_ref(_tag, storage, std::forward<visitor>(v));
-    }
-
-    static uint32_t count() { return type_info::count; }
-    void set_which( uint32_t w ) {
-      FC_ASSERT( w < count()  );
-      this->~static_variant();
-      try {
-         _tag = w;
-         impl::storage_ops<void, 0, Types...>::con(_tag, storage);
-      } catch ( ... ) {
-         _tag = 0;
-         impl::storage_ops<void, 0, Types...>::con(_tag, storage);
-      }
-    }
-
-    int which() const {return _tag;}
-
-    template<typename X>
-    bool contains() const { return which() == tag<X>::value; }
-
-    template<typename X>
-    static constexpr int position() { return impl::position<X, Types...>::pos; }
-
-    template<int Pos, std::enable_if_t<Pos < type_info::size,int> = 1>
-    using type_at = typename impl::type_at<Pos, Types...>::type;
-};
-
-template<typename Result>
-struct visitor {
+  template<typename T> void operator()( T& v )const
+  {
+      from_variant( var, v );
+  }
 };
 
-   struct from_static_variant
-   {
-      variant& var;
-      from_static_variant( variant& dv ):var(dv){}
-
-      template<typename T> void operator()( const T& v )const
-      {
-         to_variant( v, var );
-      }
-   };
-
-   struct to_static_variant
-   {
-      const variant& var;
-      to_static_variant( const variant& dv ):var(dv){}
-
-      template<typename T> void operator()( T& v )const
-      {
-         from_variant( var, v );
-      }
-   };
-
+template<typename... T> void to_variant( const std::variant<T...>& s, fc::variant& v )
+{
+  variants vars(2);
+  vars[0] = s.index();
+  visit( from_static_variant(vars[1]), s );
+  v = std::move(vars);
+}
+template<typename... T> void from_variant( const fc::variant& v, std::variant<T...>& s )
+{
+  auto ar = v.get_array();
+  if( ar.size() < 2 ) return;
+  from_index(s, ar[0].as_uint64());
+  visit( to_static_variant(ar[1]), s );
+}
 
-   template<typename... T> void to_variant( const fc::static_variant<T...>& s, fc::variant& v )
-   {
-      variant tmp;
-      variants vars(2);
-      vars[0] = s.which();
-      s.visit( from_static_variant(vars[1]) );
-      v = std::move(vars);
-   }
-   template<typename... T> void from_variant( const fc::variant& v, fc::static_variant<T...>& s )
-   {
-      auto ar = v.get_array();
-      if( ar.size() < 2 ) return;
-      s.set_which( ar[0].as_uint64() );
-      s.visit( to_static_variant(ar[1]) );
-   }
+template<typename... T> struct get_typename { static const char* name()   { return BOOST_CORE_TYPEID(std::variant<T...>).name();   } };
```

Comment:
are they equivalent, name() for std::variant and fc::static_variant?

Reply:
No they aren't.  Is that problematic?

Reply:
It doesn't look like we use `fc::static_variant` in chainbase objects, so I believe this is fine. You can double check by searching for "chainbase::object" and verifying none of the types are `fc::static_variant` or contain a `fc::static_variant`